nemo/collections/nlp/parts/microbatch_calculator.py

```diff
@@ -0,0 +1,205 @@
+from abc import ABC, abstractmethod
+from typing import List, Optional
+
+from nemo.utils import logging
+
+_GLOBAL_NUM_MICROBATCHES_CALCULATOR = None
+
+
+class NumMicroBatchesCalculator(ABC):
```

Comment:
If these are coming from apex, need to have unittest asserting exact equivalence.